apps/ai-content-generator/src/App.tsx

```diff
@@ -11,29 +11,42 @@ import Page from '@locations/Page';
 import SidebarV1 from './locations/SidebarV1';
 import Sidebar from '@locations/Sidebar';
 import Home from '@locations/Home';
+import { useFlags, useLDClient } from 'launchdarkly-react-client-sdk';
+import { ldConfigType } from '@configs/launch-darkly/ldConfig';
 
-const useV2App = import.meta.env.VITE_SHOW_AI_CONTENT_GENERATOR_V2 === 'true';
-
-const ComponentLocationSettings = {
-  [locations.LOCATION_APP_CONFIG]: useV2App ? ConfigScreen : ConfigScreenV1,
+const ComponentLocationSettings = (isV2: boolean) => ({
+  [locations.LOCATION_APP_CONFIG]: isV2 ? ConfigScreen : ConfigScreenV1,
   [locations.LOCATION_ENTRY_FIELD]: Field,
   [locations.LOCATION_ENTRY_EDITOR]: EntryEditor,
-  [locations.LOCATION_DIALOG]: useV2App ? Dialog : DialogV1,
-  [locations.LOCATION_ENTRY_SIDEBAR]: useV2App ? Sidebar : SidebarV1,
+  [locations.LOCATION_DIALOG]: isV2 ? Dialog : DialogV1,
+  [locations.LOCATION_ENTRY_SIDEBAR]: isV2 ? Sidebar : SidebarV1,
   [locations.LOCATION_PAGE]: Page,
   [locations.LOCATION_HOME]: Home,
-};
+});
 
 const App = () => {
   const sdk = useSDK();
+  const { integrationsAiContentGeneratorV2: isV2 } = useFlags<ldConfigType>();
+  const ldClient = useLDClient();
```

Comment:
Eventually it'd be nice to extract this setup into its own hook that we could put in our `marketplace-apps-toolkit` and just do something `useLaunchDarkly(sdk)` in future iterations.

Reply:
Agreed, I think it'd be nice to move this ldClient stuff out of the component